webclient/src/hooks/client.js

```diff
@@ -0,0 +1,71 @@
+import M from 'materialize-css';
+
+export async function client(endpoint, { body, ...customConfig } = {}) {
+  const token = localStorage.getItem('token');
+  const headers = { 'Content-Type': 'application/json' };
+
+  if (token) {
+    headers.Authorization = `${token}`;
+  }
+
+  const config = {
+    method: body ? 'POST' : 'GET',

```

Comment:
hmm, I think it shouldn't be designed this way, it's possible to have POST request without body (perhaps with only queries and/or params)

Reply:
it is only for request body. 
GET queries and params will be handled separately

Reply:
will add method field in customconfig to allow flexible overriding such as DELETE

Reply:
done! ^^

Reply:
okok, nice thanks :)